File: Server/ChatServer/src/main/java/com/szlazakm/chatserver/dtos/MessageAcknowledgementDTO.java
package com.szlazakm.chatserver.dtos;

import lombok.Value;
import lombok.experimental.SuperBuilder;
import lombok.extern.jackson.Jacksonized;

import java.util.UUID;

@Value
@SuperBuilder
@Jacksonized
public class MessageAcknowledgementDTO {

    UUID messageId;
    String phoneNumber;
    Long nonceTimestamp;
    byte[] nonce;
    byte[] authMessageSignature;

}

File: client/app/src/main/java/com/szlazakm/safechat/webclient/dtos/MessageAcknowledgementDTO.kt
package com.szlazakm.safechat.webclient.dtos

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.Instant
import java.util.UUID

data class MessageAcknowledgementDTO (
    @JsonProperty("messageId")
    val messageId: UUID,

    val phoneNumber: String,
    val nonceTimestamp: Long,
    val nonce: ByteArray,
    val authMessageSignature: ByteArray
)